src/client/pythonEnvironments/common/posixUtils.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as fsapi from 'fs-extra';
+import * as path from 'path';
+import { getPathEnvironmentVariable } from '../../common/utils/platform';
+
+/**
+ * Checks if a given path ends with python*.exe
+ * @param {string} interpreterPath : Path to python interpreter.
+ * @returns {boolean} : Returns true if the path matches pattern for windows python executable.
+ */
+export function isPosixPythonBin(interpreterPath:string): boolean {
+    /**
+     * This Reg-ex matches following file names:
+     * python
+     * python3
+     * python38
+     * python3.8
+     */
+    const posixPythonBinPattern = /^python(\d+(.\d+)?)?$/;
+
+    return posixPythonBinPattern.test(path.basename(interpreterPath));
+}
+
+export async function commonPosixBinPaths(): Promise<string[]> {
+    const searchPaths = (getPathEnvironmentVariable() || '')
+        .split(path.delimiter)
+        .filter((p) => p.length > 0);
+
+    const paths: Set<string> = new Set(
+        [
+            '/bin',
+            '/etc',
+            '/lib',
+            '/lib/x86_64-linux-gnu',
+            '/lib64',
+            '/sbin',
+            '/snap/bin',
+            '/usr/bin',
+            '/usr/games',
+            '/usr/include',
+            '/usr/lib',
+            '/usr/lib/x86_64-linux-gnu',
+            '/usr/lib64',
+            '/usr/libexec',
+            '/usr/local',
+            '/usr/local/bin',
+            '/usr/local/etc',
+            '/usr/local/games',
+            '/usr/local/lib',
+            '/usr/local/sbin',
+            '/usr/sbin',
+            '/usr/share',
+            '~/.local/bin',
+        ].concat(searchPaths),
+    );
+
+    const commonPaths:string[] = [];
+    // eslint-disable-next-line no-restricted-syntax
+    for (const p of paths) {
+        // eslint-disable-next-line no-await-in-loop
+        if (await fsapi.pathExists(p)) {
+            commonPaths.push(p);
+        }
+    }
```

Comment:
We don't exit early here, and we actually want to go through every item, so if you want to flex your Promise skills here's a warning-less alternative (just because we can):

```suggestion
    const promises = await Promise.all(paths.map(fsapi.pathExists));
    const commonPaths = paths.filter((_, index) => promises[index]);
```